editor/src/components/editor/store/store-deep-equality-instances.ts

```diff
@@ -1773,6 +1774,8 @@ export const ElementInstanceMetadataKeepDeepEquality: KeepDeepEqualityCall<Eleme
     ConditionValueKeepDeepEquality,
     (metadata) => metadata.textContent,
     nullableDeepEquality(StringKeepDeepEquality),
+    (m) => m.variablesInScope,
+    createCallWithShallowEquals(),
```

Comment:
the shallow equals will fail if the variableInScope's value is an object. do we filter out object values? btw we moved the element props to allElementProps exactly so that we don't blow the reference equality of the rest of metadata! it sounds like the variables in scope should also be outside of metadata, next to allElementProps

Reply:
refactored in https://github.com/concrete-utopia/utopia/pull/4496/commits/c48b9228b5d7dceb585bc9cc46e21a61cf66c7a9